docs/states.md

```diff
@@ -0,0 +1,53 @@
+# Application States (#status)
+
+
+`ApplicationChoice` uses a State Machine to manage the states the model can be in.
+
+We use a library called [workflow](https://github.com/geekq/workflow) to manage this.
+
+The workflow definition is stored in [ApplicationStateChange](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/application_state_change.rb).
+
+
+## Current States
```

Comment:
This is looking good & will be incredibly helpful! What are your views on adding an additional column to describe if the state is considered successful/unsuccessful/in-progress?

Reply:
As in, show which Status Categories they belong to? Like a reverse lookup? I think that might be useful.